scimono-server/src/main/java/com/sap/scimono/entity/bulk/validation/BulkOperationsValidator.java

```diff
@@ -89,8 +89,10 @@ public List<RequestOperation> getValidBulkOperations(BulkBody<RequestOperation>
   }
 
   public BulkBody<ResponseOperation> getValidResponseData(BulkBody<RequestOperation> bulkRequest, BulkBody<ResponseOperation> bulkResponse) {
-    Map<String, RequestOperation> requestOperations = bulkRequest.getOperations().stream()
-        .collect(Collectors.toMap(RequestOperation::getBulkId, Function.identity()));
+    Map<String, RequestOperation> requestOperations = new HashMap<>();
+    for (RequestOperation request : bulkRequest.getOperations()) {
+      requestOperations.put(request.getBulkId(), request);
+    }
```

Comment:
This is a bit better than the original code as it won't fail due to a null key collision. However, there is another issue: any non-POST request with null bulkId can replace a POST request with a null bulkId:
`requestOperations.get(respOperation.getBulkId())` could return the wrong request. 

The entire map is redundant and could be deleted: change on line 110 `if (reqOperation.getMethod() ==` to `if (respOperation.getMethod() == ` and delete the passed `reqOperation` parameter.